src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80112BE4.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/Interface_LoadItemIcon.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80112C0C.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/Item_Give.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801143CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80114978.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801149A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80114A9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80114B84.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80114CA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/Interface_HasEmptyBottle.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/Interface_HasItemInBottle.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80114FD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80115130.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801152B8.s")
+s16 sAmmoRefillCounts[] = {
+    5, 10, 20, 30, 10, 30, 40, 50, 20, 10, 1, 5, 1, 5, 10, 20, 50, 100, 200, 0,
+};
+u8 Item_Give(GlobalContext* globalCtx, u8 item) {
+    Player* player = GET_PLAYER(globalCtx);
+    u8 i;
+    u8 temp;
+    u8 slot;
+
+    slot = SLOT(item);
+    if (item >= ITEM_STICKS_5) {
+        slot = SLOT(sExtraItemBases[item - ITEM_STICKS_5]);
+    }
+
+    if (item == ITEM_SKULL_TOKEN) {
+        SET_QUEST_ITEM(item - ITEM_SKULL_TOKEN + QUEST_SKULL_TOKEN);
+        Inventory_IncrementSkullTokenCount(globalCtx->sceneNum);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_TINGLE_MAP) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BOMBERS_NOTEBOOK) {
+        SET_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK);
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_HEART_PIECE_2) || (item == ITEM_HEART_PIECE)) {
+        gSaveContext.save.inventory.questItems += (1 << QUEST_HEART_PIECE);
+        if ((gSaveContext.save.inventory.questItems & 0xF0000000) == (1 << QUEST_HEART_CONTAINER)) {
+            gSaveContext.save.inventory.questItems ^= (1 << QUEST_HEART_CONTAINER);
+            gSaveContext.save.playerData.healthCapacity += 0x10;
+            gSaveContext.save.playerData.health += 0x10;
+        }
+        return ITEM_NONE;
+
+    } else if (item == ITEM_HEART_CONTAINER) {
+        gSaveContext.save.playerData.healthCapacity += 0x10;
+        gSaveContext.save.playerData.health += 0x10;
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SONG_SONATA) && (item <= ITEM_SONG_LULLABY_INTRO)) {
+        SET_QUEST_ITEM(item - ITEM_SONG_SONATA + QUEST_SONG_SONATA);
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SWORD_KOKIRI) && (item <= ITEM_SWORD_GILDED)) {
+        SET_EQUIP_VALUE(EQUIP_SWORD, item - ITEM_SWORD_KOKIRI + 1);
```

Comment:
Yes, that will once again come into play with the enums from the OoT Equivalent:
https://github.com/zeldaret/oot/pull/1142

Based on those names, it would eventually look something like:
`SET_EQUIP_VALUE(EQUIP_SWORD, item - ITEM_SWORD_KOKIRI + EQUIP_VALUE_SWORD_KOKIRI);`

Reply:
Speaking of this, I think I'll just go ahead and add in the EQUIP_VALUE enums tbh